README.md

```diff
@@ -2,31 +2,102 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Define what a compile-time error is.
+- Define what a runtime error is.
+- Analyze the difference between the two types of errors.
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+Now that we have gotten more comfortable writing some Java code, let's discuss
+some errors we might come across when compiling and running our programs. We
+can break down these errors into two categories: compile-time errors and runtime
+errors.
 
-## Topic 1
+## What are Compile-Time Errors?
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+**Compile-time errors** are errors that occur when we try and compile the code.
+These errors won't even allow us to run the program. If we think back to "The
+Compiler and the JVM" lesson earlier in this module, there were two commands we
+used: `javac` and `java`. The `javac` command would compile the source code in
+order to turn it into something that the JVM could execute. If we try to compile
+some code, and we see errors, these are called compile-time errors as they occur
+at the time of compilation.
 
-## Topic 2
+We can see these errors when we try to run our programs in IntelliJ too.
+Remember back in the "Variables" lesson when we showed what happened if we
+attempted to use a variable _before_ it was initialized?
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+![use-variable-without-initialization-intellij](https://curriculum-content.s3.amazonaws.com/java-mod-1/variables/intellij-use-variable-without-initialization.png)
 
-## Conclusion
+We see IntelliJ warning us of an error. When we try to run the program, Java
+will first try to compile the source code. But it stops before it can even
+proceed running it:
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+![build-error](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/intellij-compile-time-error.png)
 
-## Resources
+We see that the build failed. This means, it failed to compile the code.
 
-- [Resource Link 1](example.com)
-- [Resource Link 2](example.com)
+Some examples of compile-time errors include:
+
+- Missing parentheses and curly brackets.
+
+  ![missing-curly-bracket](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-missing-curly-bracket.png)
+
+- Missing terminator.
+
+  ![missing-semicolon](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-missing-terminator.png)
+
+- And other syntax type errors.
+
+  ![syntax-errors](https://curriculum-content.s3.amazonaws.com/java-mod-1/errors/compile-time-error-syntax-errors.png)
+
```

Comment:
Consider adding an explanation of why compile-time errors are really convenient for developers.
For example, in JavaScript or Python, a missing comma or misspelled variable name can lead to an hour of frustration.
I've noticed a lot of new developers don't understand all the ways their tools and languages are there to help them succeed with minimal pain.

Reply:
Good idea!